ivy/functional/frontends/numpy/creation_routines/numerical_ranges.py

```diff
@@ -156,3 +158,14 @@ def __init__(self):
 
 
 ogrid = OGrid()
+
+
+@handle_numpy_dtype
+@to_ivy_arrays_and_back
+def geomspace(start, stop, num=50, endpoint=True, dtype=None, axis=0):
+    start, stop = promote_types_of_numpy_inputs(start, stop)
```

Comment:
I have no idea as to why but importing `promote_types_of_numpy_inputs` causes some very strange import error. You can remove this line though and the above import and the function will still pass